packages/server/watch.ts

```diff
@@ -50,7 +49,8 @@ function buildAndRun(delay: number):Operation<void> {
       yield executeAndOut('clean');
       yield executeAndOut('generate');
       yield sleep(delay);
-      let server: StdIO = daemon(scope, 'node dist/start.js');
+
+      let server: StdIO = daemon('node dist/start.js').run(scope);
```

Comment:
is there a `Runnable<T>` interface in effection?

Reply:
Not currently. The idea is that we're not keen to rush into abstractions that can be difficult to unwind, but instead let the formal abstractions brew in the applications that use them. However, when there is a pattern that is useful, such as always making it such that anything that spawns new tasks can be late-bound with a unary function that takes the parent, we want to stick to it. That way, if we do make a formal abstraction around it later, it will be eaiser to migrate towards.

Currently, a nice solution might be to have a method on `Task` called `using()` that would take an object like this, so that you could re-write the above with:

```ts
let server = scope.using(daemon('node dist/start.js'));
//just calls resource.run(self) under the covers, or something like that.
```

Reply:
```ts
let server = scope.using(daemon('node dist/start.js'));
//just calls resource.run(self) under the covers, or something like that.
```

I really, really like that.  So simple to write but doing sooo much for the dev.  I'm really excited about effection after having a look at the code and understanding it more.

One thing I was thinking is some sort of convention for disposing of things,....but that is outside the bounds of this PR :).